README.md

```diff
@@ -305,20 +305,28 @@ Makes sure kubectl is installed and available for use. Customize the version
 by specifying the `KUBECTL_VERSION` envrionmental variable. Default: `v1.4.0`.
 
 ### prepare-kubectl
-Initializes the Kubernetes config to be used with kubectl. There are 2 modes of
-operation, with an environment variable to store the config, or using `gcloud`.
-
-If `KUBECONFIG_DATA` is defined this script will `base64 --decode` and place
-the value into `~/.kube/config` to be used by kubectl.
+Initializes the Kubernetes config to be used with kubectl using a base64-encoded
+config file from the `KUBECONFIG_DATA` variable. If `KUBECONFIG_DATA` is defined
+this script will `base64 --decode` and place the value into `KUBECONFIG` to be
+used by kubectl.
 
 To generate a `KUBECONFIG_DATA` value you can use `cat ~/.kube/config | base64`.
 
-If `KUBECONFIG_DATA` is not defined then the script will check for environment
-variables used for `gcloud`. This requires:
+### prepare-gcloud
+Uses `gcloud` to download and configure `kubectl`, configures Docker to use
+Container Registry and sets the default GCP projecct. This requires:
 
 * `CLUSTER_NAME` - The short name of the cluster as shown in the GKE dashboard
 * `GCP_ZONE` - The zone of the cluster
 * `GCP_PROJECT` - The GCP project name (as passed to `gcloud` with `--project`)
+* `GCLOUD_KEY` - The base64-encoded service account credentials
+
+### prepare-awscli
+Uses `aws` to configure Docker to use ECR. This requires:
+
+* `AWS_ECR_ACCOUNT_ID` - The account ID for the ECR account to use.
+* `AWS_DEFAULT_REGION` - The region to use
+* Standard AWS credentials for `aws`
```

Comment:
Can you move these sections to be alphabetical by command name? That looks to be what the others are.